File: Java/src/main/java/Solution801_900/Solution867.java
package Solution801_900;

/**
 * @author windows
 */
public class Solution867 {
    public int[][] transpose(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;
        int[][] res = new int[m][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                res[j][i] = matrix[i][j];
            }
        }
        return res;
    }


    public static void main(String[] args) {
        Solution867 solution867 = new Solution867();
    }
}

File: C++/src/Solution801_900/Solution867.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution867.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<vector<int>> Solution::transpose(vector<vector<int>> &matrix) {
    int m = matrix.size();
    int n = matrix[0].size();
    vector<vector<int>> res(n, vector<int>(m));
    for (int i = 0; i < m; ++i) {
        for (int j = 0; j < n; ++j) {
            res[j][i] = matrix[i][j];
        }
    }
    return res;
}
